src/rgw/rgw_sal_motr.cc

```diff
@@ -443,6 +449,25 @@ int MotrUser::store_user(const DoutPrefixProvider* dpp,
     secret_key = k.key;
     MotrAccessKey MGWUserKeys(access_key, secret_key, info.user_id.to_str());
     store->store_access_key(dpp, y, MGWUserKeys);
+    access_key_tracker.insert(access_key);
```

Comment:
is there chance of access_key_tracker being empty when initialized and later someone creates first key, in this case it will not allow to add key

Try this test delete all the access keys and when there are no access keys they try adding one

Reply:
`access_key_tracker` will remain empty initially and will be filled once user is loaded. When we create a user, one set of access/secret key also gets created. So `access_key_tracker` will have that access key.

I have tested the scenario you mentioned, deleted all the keys from the user and then added a new key. I am able to see the expected results:

```
# user info
root@ssc-vm-g4-rhev4-0088 ~/cortx-rgw/build (feature/IAM/CORTX-28060)
$ ./bin/radosgw-admin user info --uid=johndoe11 --no-mon-config 
2022-03-02T03:14:16.898-0700 7fb121d155c0 -1 WARNING: all dangerous and experimental features are enabled.
{
    "user_id": "johndoe11",
    "display_name": "John Doe",
    "email": "",
    "suspended": 0,
    "max_buckets": 1000,
    "subusers": [],
    "keys": [
        {
            "user": "johndoe11",
            "access_key": "2S3OBNQWTJ0BN3WB67PJ",
            "secret_key": "gGaUsaKv3GIsliofooAjCtwKBJwB8WwiMDBceM5L"
        }
    ],
    "swift_keys": [],
    "caps": [],
    "op_mask": "read, write, delete",
    "default_placement": "",
    "default_storage_class": "",
    "placement_tags": [],
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "user_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "temp_url_keys": [],
    "type": "rgw",
    "mfa_ids": []
}

# delete access key from user, it will now have no keys
root@ssc-vm-g4-rhev4-0088 ~/cortx-rgw/build (feature/IAM/CORTX-28060)
$ ./bin/radosgw-admin key rm --uid=johndoe11 --access-key 2S3OBNQWTJ0BN3WB67PJ --no-mon-config 
2022-03-02T03:15:46.093-0700 7f74a27675c0 -1 WARNING: all dangerous and experimental features are enabled.
{
    "user_id": "johndoe11",
    "display_name": "John Doe",
    "email": "",
    "suspended": 0,
    "max_buckets": 1000,
    "subusers": [],
    "keys": [],
    "swift_keys": [],
    "caps": [],
    "op_mask": "read, write, delete",
    "default_placement": "",
    "default_storage_class": "",
    "placement_tags": [],
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "user_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "temp_url_keys": [],
    "type": "rgw",
    "mfa_ids": []
}

# create a new set of keys
root@ssc-vm-g4-rhev4-0088 ~/cortx-rgw/build (feature/IAM/CORTX-28060)
$ ./bin/radosgw-admin key create --uid=johndoe11 --gen-access-key --no-mon-config 
2022-03-02T03:15:59.271-0700 7f36691785c0 -1 WARNING: all dangerous and experimental features are enabled.
{
    "user_id": "johndoe11",
    "display_name": "John Doe",
    "email": "",
    "suspended": 0,
    "max_buckets": 1000,
    "subusers": [],
    "keys": [
        {
            "user": "johndoe11",
            "access_key": "2I4418LWUV65KE7POU63",
            "secret_key": "UtQ67Pjx7OO6qD9LWPnnTcBAqSDpkNRHuQtzxCdT"
        }
    ],
    "swift_keys": [],
    "caps": [],
    "op_mask": "read, write, delete",
    "default_placement": "",
    "default_storage_class": "",
    "placement_tags": [],
    "bucket_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "user_quota": {
        "enabled": false,
        "check_on_raw": false,
        "max_size": -1,
        "max_size_kb": 0,
        "max_objects": -1
    },
    "temp_url_keys": [],
    "type": "rgw",
    "mfa_ids": []
}

# add newly created keys to s3cmd config
root@ssc-vm-g4-rhev4-0088 ~/cortx-rgw/build (feature/IAM/CORTX-28060)
$ cat /root/.s3cfg 
access_key = 2I4418LWUV65KE7POU63
secret_key = UtQ67Pjx7OO6qD9LWPnnTcBAqSDpkNRHuQtzxCdT
host_base = localhost:8000
host_bucket = %(bucket)localhost

# perform I/O
root@ssc-vm-g4-rhev4-0088 ~/cortx-rgw/build (feature/IAM/CORTX-28060)
$ s3cmd --no-ssl mb s3://somebkt
Bucket 's3://somebkt/' created
root@ssc-vm-g4-rhev4-0088 ~/cortx-rgw/build (feature/IAM/CORTX-28060)
$ s3cmd --no-ssl ls
2022-03-02 10:19  s3://somebkt

```